jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -126,86 +126,24 @@ pointer_typet select_pointer_typet::specialize_generics(
 ///   such instantiation exists.
 optionalt<pointer_typet>
 select_pointer_typet::get_recursively_instantiated_type(
-  const irep_idt &parameter_name,
-  const generic_parameter_specialization_mapt
-    &generic_parameter_specialization_map) const
+  irep_idt parameter_name,
+  generic_parameter_specialization_mapt
+    generic_parameter_specialization_map) const
 {
-  generic_parameter_recursion_trackingt visited;
-  const size_t max_depth =
-    generic_parameter_specialization_map.find(parameter_name)->second.size();
-
-  irep_idt current_parameter = parameter_name;
-  for(size_t depth = 0; depth < max_depth; depth++)
+  while(true)
```

Comment:
Looks good, but since taking a map by value like this is quite unusual please add comments noting the strategy you're using here and why